src/App.tsx

```diff
@@ -1,24 +1,289 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+import { StatusFilter } from './types/StatusFilter';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoItem } from './components/TodoItem';
+import * as todoService from './api/todos';
+import { getFilterTodos } from './utils/getFilterTodos';
 
-const USER_ID = 0;
+const USER_ID = 11465;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [status, setStatus] = useState<StatusFilter>(StatusFilter.All);
+  const [todoError, setTodoError] = useState<ErrorType | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isRequesting, setIsRequesting] = useState(false);
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([0]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    todoService.getTodos(USER_ID)
+      .then(todoFromServer => {
+        setTodos(todoFromServer);
+        setIsLoading(false);
+      })
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.warn(error);
+        setTodoError(ErrorType.GetData);
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      setTodoError(null);
+    }, 3000);
+
+    return () => clearTimeout(timeoutId);
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    if (!todoError) {
      return;
    }
  
    const timeoutId = setTimeout(() => {
      setTodoError(null);
    }, 3000);

    return () => clearTimeout(timeoutId);
  }, [todoError]);
```
Add dependency to create timeout every time you have an error.
Also, create an `ErrorNotification` component and move this logic to it too.